src/font_loader.cpp

```diff
@@ -13,54 +23,184 @@ void ensure_unifont_loaded( std::vector<std::string> &font_list )
+unsigned int font_config::imgui_config() const
 {
-    try {
-        JsonValue json = json_loader::from_path( path );
-        JsonObject config = json.get_object();
-        if( config.has_string( "typeface" ) ) {
-            typeface.emplace_back( config.get_string( "typeface" ) );
-        } else {
-            config.read( "typeface", typeface );
-        }
-        if( config.has_string( "gui_typeface" ) ) {
-            gui_typeface.emplace_back( config.get_string( "gui_typeface" ) );
+    unsigned int ret = 0;
+    if( !antialiasing ) {
+        ret |= ImGuiFreeTypeBuilderFlags_Monochrome;
+        ret |= ImGuiFreeTypeBuilderFlags_MonoHinting;
+    }
+    if( hinting != std::nullopt ) {
+        ret |= *hinting;
+    }
+    return ret;
+}
+
+std::optional<ImGuiFreeTypeBuilderFlags> hint_to_fonthint( const std::string_view hinting )
+{
+    if( hinting == "Auto" ) {
+        return ImGuiFreeTypeBuilderFlags_ForceAutoHint;
+    }
+    if( hinting == "NoAuto" ) {
+        return ImGuiFreeTypeBuilderFlags_NoAutoHint;
+    }
+    if( hinting == "Light" ) {
+        return ImGuiFreeTypeBuilderFlags_LightHinting;
+    }
+    if( hinting == "Bitmap" ) {
+        return ImGuiFreeTypeBuilderFlags_Bitmap;
+    }
+    if( hinting == "Default" ) {
+        return std::nullopt;
+    }
+    debugmsg( "'%s' is an invalid font hinting value.", hinting );
+    return std::nullopt;
+}
+
+void font_config::deserialize( const JsonObject &jo )
+{
+    bool has_path = jo.read( "path", path, true );
+    if( !( has_path ) ) {
+        jo.throw_error( "Dummy error - force config read to return false." );
+    }
+    // Manually read hinting.
+    // Specifying enum traits for would allow all options, and we want only
+    // some to be available to the user.
+    if( jo.has_string( "hinting" ) ) {
+        hinting = hint_to_fonthint( jo.get_string( "hinting" ) );
+    }
+    jo.read( "antialiasing", antialiasing, false );
+}
+
+static void load_font_from_config( const JsonObject &config, const std::string &key,
+                                   std::vector<font_config> &typefaces )
+{
+
+    if( config.has_string( key ) ) {
+        std::string path = config.get_string( key );
+        // Migrate old font config files. Remove after 0.I
+        if( path.find( "Terminus.ttf" ) != std::string::npos ) {
+            typefaces.emplace_back( path, ImGuiFreeTypeBuilderFlags_Bitmap );
+        }  else if( path.find( "Roboto-Medium.ttf" ) != std::string::npos ) {
+            typefaces.emplace_back( path, ImGuiFreeTypeBuilderFlags_ForceAutoHint );
         } else {
-            config.read( "gui_typeface", gui_typeface );
+            typefaces.emplace_back( path );
         }
-        if( config.has_string( "map_typeface" ) ) {
-            map_typeface.emplace_back( config.get_string( "map_typeface" ) );
+    } else if( config.has_object( key ) ) {
+        // TODO: This should be doable without having to instantiate a fake path.
```

Comment:
Yes. Make a constructor with no arguments?

Reply:
I could but what do I set as the initial path value?